akka-docs/src/main/paradox/typed/persistence.md

```diff
@@ -81,11 +82,13 @@ are executed sequentially after successful execution of the persist statement (o
 
 When an event has been persisted successfully the new state is created by applying the event to the current state with the `eventHandler`.
 
-The event handler returns the new state, which must be immutable so you return a new instance of the state.
+The state is typically immutable and then the event handler returns a new instance of the state. If the state is
+mutable the event handler may update the state instance and return the same instance.
+
 The same event handler is also used when the entity is started up to recover its state from the stored events.
 
-It is not recommended to perform side effects
-in the event handler, as those are also executed during recovery of an persistent actor
+It is not recommended to perform side effects in the event handler, as those are also executed during recovery of
+an persistent actor
```

Comment:
Should this sentence have stronger wording about not doing side effects?

Reply:
rephrase